### ISSUE TITLE
Pluck/waveguide oscillator

### ISSUE THREAD

**Issue by baconpaul** [surge-synthesizer/surge#894]
- [x] Absolute mode for offset
- [x] "First Cycle" not in filter properly so run it out after loading delay buffer (or run the pre-fill across the filter perhaps?)
- [x] Extend range on string 2 detune doesn't work
- [x] ~Crash when loading a skin that has a label like this `<label x="350" y="30" w="50" h="20" control_text="osc.param_1"/>`~
- [x] Tone left/right is linear vs exponential and so make it more uniform
- [x] Do some profiling

**Comment by mortfell** [comment]
Was this something Claes started but never finished??
Is it a physical model or something?

**Comment by baconpaul** [comment]
Yes
Presume so. Some sort of KS variant I bet. There’s really almost no code there for this one (unlike reverb2 where a lot of it is finished).

**Comment by mortfell** [comment]
Oh interesting! Yeah I love the different OSC algorithm thing. The FM modes are one of my favorite things about this synth.

Would it be possible/reasonable given Surge's DSP code and structure to use code from one of the mutable instruments Plaits KS type algorithms? Or any of the other ones? Since they are open-source?

I feel like a lot of them would be kinda redundant since Surge already covers a lot of those sounds with much more options, but the physical model things and the formant filter could be neat as Surge Oscillators

**Comment by baconpaul** [comment]
Ha yeah I suppose we could. It’s all gpl3. Those are cool

And then we could get surge rack including mutable in rack with a different panel than mutable 

I would ask first if I did this of course

**Comment by baconpaul** [comment]
(I’ve also thought about including some 8-bit and chip-tune type oscillators also; for rack I wrote a pretty good simulation of the NES wave and noise generator and could add those easily too. I’ve also been kicking around the idea of a phase modulation oscillator too to complement the two FM ones. And of course don’t forget anything can FM anything else using the FM routing; and the sin oscillators have the various quadrant masking tricks like the TX-81Z since 1.6.1).

**Comment by mortfell** [comment]
Yeah the FM options are very great overall in Surge. I have been designing a couple "FX" patches using the one shot wav sample feature to load short vocal samples that FM more tonal sounding oscillators. Not very musical, but really cool sounding.

That's really neat, more OSC types in the mix would be great in this synth!

I should read up on/experiment with the the quadrant masking. I have heard about how that synth was one of the first commercial FM implementations to allow the use of Waveforms other than plain sine waves. Is that how it did that?

**Comment by mortfell** [comment]
Also, I realize now that I have used your VCV Rack stuff! I noticed the "Bacon" music tag when I last opened it haha. QuantEyes is in my default starting template! and I always love chiptune stuff, have used the noise and waves a number of times in patches.

**Comment by baconpaul** [comment]
Ha neat! Glad you find them useful. You saw surge is all one ack also broken down into modules?

Yeah if you open the surge sin oscillator and drag the waveshaper you can see a collection of waves which are basically quadrant swapping or masking or frequency doubling. Basically if you know sin and cos you can get all those shapes with simple masks and multiplies. And indeed the tx81z was one of the first fm synths to do fm on something other than sin waves which is one of the ways it got such dense sounds with a lower osc count and smaller algo space than the dx7.

**Comment by mortfell** [comment]
Yeah I've messed around with the VCV Rack surge racks, I should spend more time with them!!
I spent a lot of time with VCV Rack when I first heard about it.

Most of my synth messing around time has been dedicated to Surge and Serum lately, they have been my favorite synths as of late, and as it's easier to use sounds I design in them for production and songwriting (which is how I always justify the amount of time I spend messing with synths 🤣to myself). Once there is a VST version of VCV Rack I will definitely spend a lot more time with it.
